data-processing/entities/sentences_pdf/commands/fetch_spp_jsons.py

```diff
@@ -0,0 +1,76 @@
+import os
+
+import json
+import time
+from argparse import ArgumentParser
+import logging
+from typing import Iterator, Optional
+from dataclasses import dataclass
+
+from common import file_utils
+from common import directories
+from common.commands.base import ArxivBatchCommand
+from common.fetch_arxiv import FetchFromArxivException, fetch_from_arxiv, fetch_from_s3
+from common.types import ArxivId, RelativePath
+
+
+import requests
+
+#  TODO - dont need
+# DEFAULT_S3_ARXIV_SOURCES_BUCKET = "s2-arxiv-sources"
+
+"""Constants for dealing with retries and delays between call attempts"""
+DEFAULT_FETCH_DELAY = 10  # seconds
+BACKOFF_FETCH_DELAY = 60  # seconds
+MAX_FETCH_ATTEMPTS = 3
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class SppJsonTask:
+    arxiv_id: ArxivId
+    pdf_path: RelativePath
+    spp_host: str
```

Comment:
We are now passing in two env vars when the data processor is invoked:

1) `SPP_HOST`
2) `SPP_PORT`

Reply:
From my foggy memory of the `Command` interface, there's a class method you can add to grab env vars for use during command execution.